docs/docs/validating-plugin-options.md

```diff
@@ -0,0 +1,230 @@
+
+To verify that a `pluginOptionsSchema` behaves as expected, unit test it with different configurations using the [`gatsby-plugin-utils` package](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-utils#testpluginoptionsschema).
+
+1. Add the `gatsby-plugin-utils` package to your site:
+
+   ```shell
+   yarn add gatsby-plugin-utils
+   ```
+
+   or
+
+   ```shell
+   npm install gatsby-plugin-utils
+   ```
+
+2. Use the `testPluginOptionsSchema` function exported from it in your test file. For example, with [Jest](https://jestjs.io):
+
+   ```javascript:title=plugins/gatsby-plugin-console/__tests__/gatsby-node.test.js
+   // This is an example using Jest (https://jestjs.io/)
+   import { testPluginOptionsSchema } from "gatsby-plugin-utils"
+   import { pluginOptionsSchema } from "../gatsby-node"
+
+   describe(`pluginOptionsSchema`, () => {
+     it(`should invalidate incorrect options`, () => {
+       const options = {
+         optionA: undefined, // Should be a boolean
+         message: 123, // Should be a string
+         optionB: `not a boolean`, // Should be a boolean
+       }
+       const { isValid, errors } = testPluginOptionsSchema(
+         pluginOptionsSchema,
+         options
+       )
+
+       expect(isValid).toBe(false)
+       expect(errors).toEqual([
+         `"optionA" is required`,
+         `"message" must be a string`,
+         `"optionB" must be a boolean`,
+       ])
+     })
+
+     it(`should validate correct options`, () => {
+       const options = {
+         optionA: false,
+         message: "string",
+         optionB: true,
+       }
+       const { isValid, errors } = testPluginOptionsSchema(
+         pluginOptionsSchema,
+         options
+       )
+
+       expect(isValid).toBe(true)
+       expect(errors).toEqual([])
+     })
+   })
+   ```
+
+## Best practices for option schemas
+
+The [Joi API documentation](https://joi.dev/api/) is a great reference to use while working on a `pluginOptionsSchema` as it shows all the available types and methods.
+
+Here are some specific Joi best practices for `pluginOptionsSchema`s:
+
+- [Add descriptions](#add-descriptions)
+- [Set default options](#set-default-options)
+- [Validate external access](#validate-external-access) where necessary
+- [Add custom error messages](#add-custom-error-messages) where useful
+- [Deprecate options](#deprecating-options) in a major version release rather than removing them
+
+### Add descriptions
+
+Make sure that every option and field has a `.description()` explaining its purpose. While this is not immediately visible, there might be tooling in the future that auto-generates plugin option documentation from the schema.
+
+### Set default options
+
+You can use the `.default()` method to default a value for an option. For example with `gatsby-plugin-console-log`, this is how it can default the `message` option to `"default message"` in case users do not pass their own `message`:
```

Comment:
```suggestion
You can use the `.default()` method to set a default value for an option. For example with `gatsby-plugin-console-log`, you can have the `message` option default to  `"default message"` in case a user does not pass their own `message`:
```